test/cloud-canary/mzcompose.py

```diff
@@ -0,0 +1,135 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import os
+import ssl
+import subprocess
+import time
+from pathlib import Path
+from textwrap import dedent
+
+import pg8000
+
+from materialize.mzcompose import Composition, _wait_for_pg
+from materialize.mzcompose.services import Materialized, Testdrive
+
+SERVICES = [Materialized(), Testdrive()]
+
+REGION = "aws/us-east-1"
+ENVIRONMENT = "staging"
+USERNAME = "infra+nightly-canary@materialize.com"
+APP_PASSWORD = os.environ["NIGHTLY_CANARY_APP_PASSWORD"]
+VERSION = "unstable-" + os.environ["BUILDKITE_COMMIT"]
+
+
+def disable_region() -> None:
+    print(f"Shuting down region {REGION} ...")
+
+    subprocess.run(
+        ["bin/ci-builder", "run", "stable", "bin/mz", "region", "disable", REGION]
+    )
+
+
+def workflow_default(c: Composition) -> None:
+    """Deploy the current source to the cloud and run a smoke test."""
+
+    print("Obtaining mz_version() string from local instance ...")
+    c.start_and_wait_for_tcp(services=["materialized"])
+    c.wait_for_materialized()
+    local_version = c.sql_query("SELECT mz_version();")[0][0]
+
+    print("Prepare configuration file for bin/mz")
+    mz_profiles = Path.home() / ".config" / "mz" / "profiles.toml"
+    mz_profiles.parent.mkdir(parents=True, exist_ok=True)
+    with open(mz_profiles, "w+") as o:
+        o.write(
+            dedent(
+                f"""
+                current_profile = 'default'
+                [profiles.default]
+                email = '{USERNAME}'
+                app-password = '{APP_PASSWORD}'
+                region = '{REGION}'
+                endpoint = 'https://{ENVIRONMENT}.cloud.materialize.com/'
+                """
+            )
+        )
```

Comment:
I think it is a big smell to have an mzcompose workflow writing to the home directory! I have a `~./config/mz/profiles.toml` that I very much like and would not want running this composition locally to overwrite. 

Could we instead:

  * Add a mzbuild.yml to src/mz that builds a Dockerfile that builds `mz` and sets `mz` as the command.
  * Use the `c.run("mz", persistent=True)` thing to start up a dedicated `mz` container and configure the profile just inside of _that_ container?
  * I'm pro keeping your modification to ci-builder because I think it is generally good to forward `mz` credentials into the CI builder, since sometimes folks run the CI builder when developing locally since it has the Rust toolchain built in.